app/models/user.rb

```diff
@@ -83,6 +85,27 @@ def report_registrable_months
     months
   end
 
+  def report_registrable_dates
+    [*(entry_date..Date.today)]
+  end
+
+  def selectable_years(target_date)
+    [*(selectable_first_year(target_date)..selectable_last_year(target_date))]
+  end
+
+  def selectable_months(target_date)
+    months = [*(selectable_first_month(target_date)..selectable_last_month(target_date))]
+    target_day = target_date.day
+    return months - [2, 4, 6, 9, 11] if target_day == 31
+    return months - [2] if target_day == 30
```

Comment:
__[reek]__
* [FeatureEnvy] refers to 'months' more than self (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md)
* [FeatureEnvy] refers to 'target_day' more than self (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md)

<!-- Comment by [SideCI](https://sideci.com) -->